### PR TITLE
Component tabs

### PR DESCRIPTION
**Pull request by boxadesign**
### What is the context of this PR?
Describe what you have changed and why, link to other PRs or Issues as appropriate.

### How to review 
Describe the steps required to test the changes (include screenshots if appropriate).

### CODE REVIEW

**Comment by armstrongb** [comment]
@boxadesign can we amend so that the keyboard navigation requires 'enter' to activate tabs?

File: components/tabs/tabs.hbs [hunk 1]
```diff
@@ -0,0 +1,43 @@
+
+<section role="region" class="tabs u-mt-l" aria-label="This is an area of tabbed content">
+
```
Comment (armstrongb) [inline components/tabs/tabs.hbs]:
```suggestion
<h2 class="neptune u-vh u-no-js-show">{{tabs_contents_title}}</h2>
```

File: components/tabs/tabs.scss [hunk 2]
```diff
@@ -0,0 +1,95 @@
+.tabs {
+  margin-bottom: 1rem;
+
+  .no-js & { // No js display the aria-label from the tab region as content
+    &:before {
```
Comment (armstrongb) [inline components/tabs/tabs.scss]:
```suggestion
    &:before { <-- this section can be removed as we will use the heading instead.
```

File: components/tabs/tabs.hbs [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    <ul class="tabs__list">
+      <li id="tab1Item" class="tab__list-item"><a href="#tabId1" class="tab">First tab label</a></li>
```
Comment (armstrongb) [inline components/tabs/tabs.hbs]:
```suggestion
      <li id="tab1Item" class="tab__list-item"><a href="#tabId1" class="tab" data=ga="click" data-ga-category="tabs" data-ga-action="Show: {{tab}}" data-ga-label="Show: {{tab}}">First tab label</a></li>
```
or

```suggestion
 {{#each tabs}}
    <li id="tabID{{@index}}Item" class="tab__list-item"><a href="#tabID{{@index}}" class="tab" data=ga="click" data-ga-category="tabs" data-ga-action="Show: {{tab}}" data-ga-label="Show: {{tab}}">{{tab}}</a></li>
  {{/each}}
```

File: components/tabs/tabs.hbs [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+      <li id="tab2Item" class="tab__list-item"><a href="#tabId2" class="tab">Second tab</a></li>
+      <li id="tab3Item" class="tab__list-item"><a href="#tabId3" class="tab">Tab 3</a></li>
+      <li id="tab4Item" class="tab__list-item"><a href="#tabId4" class="tab">Fourth</a></li>
+    </ul>
+
+  <section id="tabId1" class="tabs__panel">
```
Comment (armstrongb) [inline components/tabs/tabs.hbs]:
```suggestion
  {{#each tabs}}
  <section id="tabID{{@index}}" class="tabs__panel">
  {{tabContent}}
  </section>
```

**Comment by boxadesign** [comment]
@armstrongb Changes have been made and tests are now written. Ready for full review.

File: components/tabs/tabs.js [hunk 5]
```diff
@@ -0,0 +1,275 @@
...
+            this.makeTabs()
+        } else {
+            this.makeList()
+        }
+    }
+
+    makeTabs() {
+        let component = this.component
+        let tabs = this.tabs
+        let tabList = component.getElementsByClassName(classTabList)
```
[the remark this answers is not in the document]
Comment (boxadesign) [reply components/tabs/tabs.js]: Very nice thanks!

File: components/tabs/tabs.js [hunk 6]
```diff
@@ -0,0 +1,275 @@
...
+        let tabListItems = component.getElementsByClassName(classTabListItems)
+
+        if (!tabs || !tabList || !tabListItems) {
+            return
+        }
+
+        tabList[0].setAttribute('role', 'tablist')
+        tabList[0].classList.remove(this.jsTabListAsListClass)
+
+        forEach(tabListItems, (item) => {
```
[the remark this answers is not in the document]
Comment (boxadesign) [reply components/tabs/tabs.js]: Yep will do

File: components/tabs/tabs.js [hunk 7]
```diff
@@ -0,0 +1,275 @@
...
+        this.hidePanel(tab)
+    }
+
+    showTab(tab) {
+        this.highlightTab(tab)
+        this.showPanel(tab)
+    }
+
+    getTab(hash) {
+        return this.component.querySelector('.tab[href="' + hash + '"]')
```
[the remark this answers is not in the document]
Comment (boxadesign) [reply components/tabs/tabs.js]: Good one. Will implement!

File: components/tabs/tabs.scss [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+    position: relative;
+    width: auto;
+    height: auto;
+    margin: 0 0 1rem;
+  }
+}
+
+// Tabs - list
+.tabs__list {
+  margin: 0;
```
Comment (armstrongb) [inline components/tabs/tabs.scss]:
```suggestion
  margin: 0;
  padding-left: 0;
```

Comment (boxadesign) [reply components/tabs/tabs.scss]: I've kept a small amount of padding on the left as it didn't look quite right when the first tab was not selected. Let me know what you think.

Comment (armstrongb) [reply components/tabs/tabs.scss]: @RhodGriffith review the PR when you get a minute as this was from your prototypes. Ta.

Comment (RhodGriffith) [reply components/tabs/tabs.scss]: @boxadesign I think it looks better without any padding on the left, can we remove this? 
Padding should be 0.5rem between tabs (which is the orange area in the screenshot). I think it's inheriting additional padding from somewhere, can we close that space to make it a bit more compact?
screen shot 2018-11-06 at 12 57 00

Comment (boxadesign) [reply components/tabs/tabs.scss]: Yep fair enough. Will push up a change in a mo

**Comment by armstrongb** [comment]
Adding 'Do not merge' until DAC audit has been performed.

**Comment by boxadesign** [comment]
@armstrongb This build includes the DAC changes.

**Comment by armstrongb** [comment]
@boxadesign the Travis build has failed.
Have we got unit tests ready for tabs too?

**Comment by boxadesign** [comment]
@armstrongb Yeah there are unit tests and they are failing because safari on ios runs the tests for tabs on desktop...so obviously fails. There are tests written specifically for mobile viewports to get good test coverage. 

Looking at it with Shane now.

**Comment by armstrongb** [comment]
DAC audit issues resolved so ready for merging.

**Pull request merged**